sql/moz-fx-data-shared-prod/firefox_desktop_derived/urlbar_events_v1/query.sql

```diff
@@ -37,154 +33,90 @@ WITH events_unnested AS (
-          'inaccurate_location',
-          'not_interested',
-          'not_relevant',
-          'show_less_frequently'
-        )
-      ),
-      TRUE
-    ) AS is_terminal,
-    COUNTIF(
-      res = 'default_partner_search_suggestion'
-    ) AS num_default_partner_search_suggestion_impressions,
-    COUNTIF(res = 'search_engine') AS num_search_engine_impressions,
-    COUNTIF(res = 'trending_suggestion') AS num_trending_suggestion_impressions,
-    COUNTIF(res = 'history') AS num_history_impressions,
-    COUNTIF(res = 'bookmark') AS num_bookmark_impressions,
-    COUNTIF(res = 'open_tabs') AS num_open_tab_impressions,
-    COUNTIF(res = 'admarketplace_sponsored') AS num_admarketplace_sponsored_impressions,
-    COUNTIF(res = 'navigational') AS num_navigational_impressions,
-    COUNTIF(res = 'add_on') AS num_add_on_impressions,
-    COUNTIF(res = 'wikipedia_enhanced') AS num_wikipedia_enhanced_impressions,
-    COUNTIF(res = 'wikipedia_dynamic') AS num_wikipedia_dynamic_impressions,
-    COUNTIF(res = 'weather') AS num_weather_impressions,
-    COUNTIF(res = 'quick_action') AS num_quick_action_impressions,
-    COUNTIF(res = 'pocket_collection') AS num_pocket_collection_impressions,
-    -- can't group by array-valued column. Instead "aggregate" by taking a single value
-    ANY_VALUE(experiments) AS experiments
-  FROM
-    events_unnested,
-    UNNEST(product_results) AS res
-  GROUP BY
-    submission_date,
-    sample_id,
-    client_id,
-    legacy_telemetry_client_id,
-    event_timestamp,
-    normalized_channel,
-    normalized_country_code,
-    event_name,
-    session_action_type,
-    engaged_result_type,
-    product_engaged_result_type,
-    first_result_type,
-    product_first_result_type,
-    num_chars_typed,
-    num_total_results,
-    annoyance_signal_type,
-    is_terminal
 )
-SELECT
-  submission_date,
-  client_id AS glean_client_id,
-  legacy_telemetry_client_id,
-  sample_id,
-  event_timestamp,
-  normalized_channel,
-  normalized_country_code,
-  event_name,
-  session_action_type,
-  engaged_result_type,
-  product_engaged_result_type,
-  annoyance_signal_type,
-  first_result_type,
-  product_first_result_type,
-  num_chars_typed,
-  num_total_results,
-  is_terminal,
-  num_default_partner_search_suggestion_impressions,
-  num_search_engine_impressions,
-  num_trending_suggestion_impressions,
-  num_history_impressions,
-  num_open_tab_impressions,
-  num_bookmark_impressions,
-  num_admarketplace_sponsored_impressions,
-  num_navigational_impressions,
-  num_add_on_impressions,
-  num_wikipedia_enhanced_impressions,
-  num_wikipedia_dynamic_impressions,
-  num_weather_impressions,
-  num_quick_action_impressions,
-  num_pocket_collection_impressions,
-  experiments
+SELECT 
+  *,
+  `mozfun.norm.result_type_to_product_name`(selected_result) AS product_selected_result,
+  enumerated_array(ARRAY(
+    SELECT
+      `mozfun.norm.result_type_to_product_name`(x)
+    FROM
+      UNNEST(SPLIT(results, ',')) AS x
```

Comment:
split is no longer needed here since `results` is already an array.

However, what about putting the product mapping step into the `enumerated_array` function, so that it returns a struct with 3 elements: offset, result type, product result type

Reply:
done - I combined it into a single enumerated_array with 3 elements